en/docs/develop/writing-a-custom-federated-authenticator.md

```diff
@@ -106,1262 +111,118 @@ public interface ApplicationAuthenticator extends Serializable {
 This API can be used to configure a custom authenticator. As an example,
 a Twitter authenticator is configured using the above API.
 
-### Writing a custom authenticator for Twitter
+### Writing a custom federated authenticator
 
-The WSO2 Identity Server comes with several built in authenticators like
-Facebook, Google, OpenID, and SAML. This topic provides instructions on
-how to authenticate users via Twitter using the SAML configurations
-available in the Identity Server.
+1. First create a maven project for the custom federated authenticator. Refer the [pom.xml](https://github.com/GANGANI/custom-federated-authenticator/blob/main/pom.xml) 
+   file used for the sample custom federated authenticator.
+2. Refer the [service component class](https://github.com/GANGANI/custom-federated-authenticator/blob/main/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/internal/CustomFederatedAuthenticatorServiceComponent.java) 
+   as well since the authenticator is written as an OSGI service to deploy in the WSO2 Identity Server and register 
+   it as a federated authenticator
+3. The [custom federated authenticator](https://github.com/GANGANI/custom-federated-authenticator/blob/main/src/main/java/org/wso2/carbon/identity/custom/federated/authenticator/CustomFederatedAuthenticator.java) 
+   should be written by extending the [AbstractApplicationAuthenticator](https://github.com/wso2/carbon-identity-framework/blob/v5.18.187/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java) class 
+   and implementing the [FederatedApplicationAuthenticator](https://github.com/wso2/carbon-identity-framework/blob/v5.18.187/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/LocalApplicationAuthenticator.java) class.
+4. You can find a custom federated authenticator from [here](https://github.com/GANGANI/custom-federated-authenticator) for your reference
```

Comment:
```suggestion
4. You can find a custom federated authenticator [here](https://github.com/GANGANI/custom-federated-authenticator) for your reference
```

Reply:
Fixed https://github.com/wso2/docs-is/pull/2527/commits/ffc80dd6ac9088892baa7c56e5f03a9aa0768d3f